File: android/src/main/java/com/rivereactnative/RNLoopMode.kt
package com.rivereactnative

import app.rive.runtime.kotlin.core.Loop as RiveLoop

enum class RNLoopMode(private val mValue: String) {
  OneShot("oneShot"),
  Loop("loop"),
  PingPong("pingPong"),
  Auto("auto");

  override fun toString(): String {
    return mValue
  }

  companion object {
    fun mapToRNLoopMode(loopMode: String): RNLoopMode {
      return valueOf(values().first() { it.toString() == loopMode }.name)
    }

    fun mapToRNLoopMode(riveLoopMode: RiveLoop): RNLoopMode {
        return when (riveLoopMode) {
          RiveLoop.ONESHOT -> OneShot
          RiveLoop.LOOP -> Loop
          RiveLoop.PINGPONG -> PingPong
          RiveLoop.AUTO -> Auto
        }
      }

    fun mapToRiveLoop(rnLoopMode: RNLoopMode): RiveLoop {
      return when (rnLoopMode) {
        OneShot -> RiveLoop.ONESHOT
        Loop -> RiveLoop.LOOP
        PingPong -> RiveLoop.PINGPONG
        Auto -> RiveLoop.AUTO
      }
    }
  }
}

File: ios/RNLoopMode.swift
enum RNLoopMode: String {
    case OneShot = "oneShot"
    case Loop = "loop"
    case PingPong = "pingPong"
    case Auto = "auto"

    static func mapToRNLoopMode(value: String) -> RNLoopMode {
        if let rnEnum = RNLoopMode(rawValue: value) {
            return rnEnum
        } else {
            // Return a default value instead of crashing
            RCTLogWarn("Unsupported loop mode type: \(value), defaulting to Auto")
            return .Auto
        }
    }


    static func mapToRNLoopMode(value: Int) -> RNLoopMode {
        if let riveEnum = RiveRuntime.RiveLoop(rawValue: value) {
            switch (riveEnum) {
            case .oneShot:
                return .OneShot
            case .loop:
                return .Loop
            case .pingPong:
                return .PingPong
            case .autoLoop:
                return .Auto
            default:
                return .Auto
            }


        } else {
            // Return a default value instead of crashing
            RCTLogWarn("Unsupported loop mode type: \(value), defaulting to Auto")
            return .Auto
        }
    }

    static func mapToRiveLoop(rnLoopMode: RNLoopMode) ->  RiveLoop {
        switch rnLoopMode {
        case .OneShot:
            return .oneShot
        case .Loop:
            return .loop
        case .PingPong:
            return .pingPong
        case .Auto:
            return .autoLoop
        }
    }
}
